logs_ingest/main.py

```diff
@@ -160,3 +162,17 @@ def parse_record(record: Dict, self_monitoring: SelfMonitoring):
 def extract_cloud_log_forwarder(parsed_record):
     if cloud_log_forwarder:
         parsed_record["cloud.log_forwarder"] = cloud_log_forwarder
+
+
+def parse_to_json(text):
+    try:
+        event_json = json.loads(text)
+    except Exception:
+        event_json = json.loads(text.replace("\'", "\""))
+    return event_json
+
+
+def convert_date_format(record):
+    timestamp = record.get("timestamp", None)
+    if timestamp and re.findall('[0-9]{2}/[0-9]{2}/[0-9]{4} [0-9]{2}:[0-9]{2}:[0-9]{2}', timestamp):
+        record["timestamp"] = str(datetime.strptime(timestamp, '%m/%d/%Y %H:%M:%S').isoformat()) + "Z"
```

Comment:
What is the "Z" for at the end?

Reply:
It means that it is UTC time (in ISO 8601 format). To be precisely it means offset from UTC. At the end of datetime might be value +01:00 that means timezone is +1, -01:00 that means timezone is -1, or Z which means no offset